sdk/core/azure-core-test/src/main/java/com/azure/core/test/InterceptorManager.java

```diff
@@ -62,6 +70,12 @@ public class InterceptorManager implements AutoCloseable {
     // Stores a map of all the HTTP properties in a session
     // A state machine ensuring a test is always reset before another one is setup
     private final RecordedData recordedData;
+    private final boolean enableTestProxy;
+    private TestProxyRecordPolicy testProxyRecordPolicy;
+    private TestProxyPlaybackClient testProxyPlaybackClient;
+    private final Queue<String> proxyVariableQueue = new LinkedList<>();
```

Comment:
Do we want to think about making this a `ConcurrentLinkedQueue` to support parallel requests?

Reply:
This is for storing variables from `testResourceNamer`. It is per-test case so there shouldn't be any parallel requests.

Reply:
Storage has tests with parallel requests for parallel upload, though I don't know if those are recorded but there are cases where this can happen

Reply:
Discussed this offline, we'll do something abouut this if it turns out to be problematic.